### DISCUSSION TITLE
Pixel editor undo option

### DISCUSSION THREAD

**Discussion by mackmind** [sn4k3/UVtools#199] [Ideas]
Hi,
is there an option to add an "undo" button for a whole brush stroke (last drawn region even if it goes through many layers) in pixel editor?

I found out that ctrl+z is not working for that. If you applied a lots of strokes on many layers, then for ex. forget to change layers above or below attribute and do a bad stroke, you can either dismiss all changes and start again or accept the bad stroke. It is quite hard to correct a bad stroke at the moment. Adding an undo button would make the pixel editor tool much more forgiving. Thanks!

**Comment by sn4k3** [comment]
Ctrl + Z is only for applied operations, including pixel drawing. While you are drawing it was not applied yet.
A stroke does not exists on UVtools, each time you move the mouse it draw a new pixel, so theres no start nor end range.
You will have to go on list, select all the range you draw and hit Remove. Since  the pixels are hightlight at green and the newest are put on top you can easy spot them.

![image](https://user-images.githubusercontent.com/113281/117684626-141f8280-b1ad-11eb-8323-86e18997b360.png)

Another strategy is to keep applying modifications so on next one if you mess anything you can just hit Clear
